test/test.c

```diff
@@ -3001,16 +3004,37 @@ static hb_dict_t * PreparePreset(const char *preset_name)
             return NULL;
         }
     }
-    else
+    else if (preset_import_gui == 1)
     {
         preset = hb_presets_get_default();
+        if (preset == NULL)
+        {
+            fprintf(stderr, "Error loading presets! Aborting.\n");
+            return NULL;
+        }
+        preset = hb_value_dup(preset);
     }
-    if (preset == NULL)
+    else
     {
-        fprintf(stderr, "Error loading presets! Aborting.\n");
-        return NULL;
+        // Default settings
```

Comment:
My recommendation here would be to create a json string that completely describes the default that you want and parse it into the preset dict here (see hb_value_json).  This makes everything a bit more explicit.